tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -75,6 +75,11 @@ class AppFilterImpl(
     val filterAppName = appArgs.applicationName.getOrElse("")
     val filterCriteria = appArgs.filterCriteria.getOrElse("")
     val userName = appArgs.userName.getOrElse("")
+    val logicFilter = if (appArgs.any()) {
```

Comment:
We have 2 file based filters 1. based on fileName match and 2. N-newest/oldest. I don't think we need to make changes to the file based filter code. I will add test which will test N-newest-filesystem along with appFilters with `any` and `all`

Reply:
yes so why don't we need to make changes.  The file based ones apply first right now and then the other user, app name, etc..  So if you specify any wouldn't you want it to match either file name or on an application name?  But if we don't change it will first filter only those filename and then apply any within those

Reply:
My understanding was we always do file based filters first followed by app based filters(i.e result of file based filter is input to app filters). But what you mentioned above makes sense for `any` case. Thanks for the use case. I will make the changes to reflect both file and app based filters.